### PR TITLE
Improves integrations page

### PR DESCRIPTION
**Pull request by andrerpbts**
This PR improves the integrations settings page, moving all logic to handle those integrations to `integrations_controller` (this is being removed from the project edit form).

captura de tela 2016-12-01 as 17 22 37
captura de tela 2016-12-01 as 17 22 57
captura de tela 2016-12-01 as 17 29 45

### CODE REVIEW

File: app/controllers/integrations_controller.rb [hunk 1]
```diff
@@ -44,5 +39,14 @@ def set_project
     @project = policy_scope(Project).friendly.find(params[:project_id])
   end
 
+  def set_integrations
+    @integrations ||= begin
+      current_integrations = @project.integrations
+      valid_integrations = Integration::VALID_INTEGRATIONS
+      missing_integrations = valid_integrations - current_integrations.pluck(:kind)
+
+      current_integrations |=  missing_integrations.map { |i| Integration.new kind: i }
```
Comment (daniloisr) [inline app/controllers/integrations_controller.rb]: I think that you don't need the assigner `|=` here, you can use just `+`.

Also, I would remove variable `valid_integrations` in line 45 and use `Integration::VALID_INTEGRATIONS` directly in 46, but this is just a suggestion to remove extra lines.

File: app/views/integrations/index.html.erb [hunk 2]
```diff
@@ -1,7 +1,53 @@
...
+  <div class="project_nav">
+    <%= link_to t('back'), project_path(@project), class: "btn btn-primary btn-sm" %>
+  </div>
 <% end %>
 
 <div class="row">
-  <%= render 'form' %>
+  <div class="col-sm-12 col-md-12">
+    <h3><%= t('projects.integrations') %></h3>
+    <hr />
```
Comment (daniloisr) [inline app/views/integrations/index.html.erb]: I know that this can be a personal taste, but you can use just `<hr>` since HTML 4.0

Comment (andrerpbts) [inline app/views/integrations/index.html.erb]: Yep, old habits. 

I'll change it.

**Comment by akitaonrails** [comment]
One thing. Leave the default value for the private Uri as the constant Integration_mattwrmost Uri as it automatically fetches the Uri from env.

**Pull request merged**